src/app/context/wallet/BTCWalletProvider.tsx

```diff
@@ -175,21 +179,18 @@ export const BTCWalletProvider = ({ children }: PropsWithChildren) => {
         btcWalletProvider?.signPsbts(psbtsHexes) ?? [],
       getNetwork: async () =>
         btcWalletProvider?.getNetwork() ?? ({} as Network),
-      signMessage: async (
-        message: string,
-        type: "ecdsa" | "bip322-simple" = "bip322-simple",
-      ) => btcWalletProvider?.signMessage(message, type) ?? "",
-      getBalance: async () => btcWalletProvider?.getBalance() ?? 0,
-      getNetworkFees: async () =>
-        btcWalletProvider?.getNetworkFees() ?? ({} as Fees),
-      pushTx: async (txHex: string) => btcWalletProvider?.pushTx(txHex) ?? "",
+      signMessage: async (message: string, type: "ecdsa") =>
+        btcWalletProvider?.signMessage(message, type) ?? "",
+      getBalance: async () => getAddressBalance(address),
+      getNetworkFees: async () => getNetworkFees(),
+      pushTx: async (txHex: string) => pushTx(txHex),
       getUtxos: async (address: string, amount?: number) =>
```

Comment:
I'm not fully understand why do we receive `address` via argument here, but for `getBalance` method we use closure

Reply:
I did not change the calling stuff
Your message is correct, you can check out the commit https://github.com/babylonlabs-io/simple-staking/pull/515/commits/e6c8c6cb0073e3e3489d96ab987ce19b63cb4f94